src/blueprint.zig

```diff
@@ -36,6 +40,44 @@ pub const Blueprint = struct {
 	pub fn clone(self: *Blueprint, allocator: NeverFailingAllocator) Blueprint {
 		return .{.blocks = self.blocks.clone(allocator)};
 	}
+	pub fn rotateZ(self: Blueprint, allocator: NeverFailingAllocator, angle: Degrees) Blueprint {
+		var new = Blueprint{
+			.blocks = switch(angle) {
+				.@"0", .@"180" => .init(allocator, self.blocks.width, self.blocks.depth, self.blocks.height),
+				.@"90", .@"270" => .init(allocator, self.blocks.depth, self.blocks.width, self.blocks.height),
+			},
+		};
+
+		const a: f32 = (std.math.pi/2.0)*@as(f32, @floatFromInt(@intFromEnum(angle)));
+		const sin: f32 = @sin(a);
+		const cos: f32 = @cos(a);
+
+		for(0..self.blocks.width) |i| {
+			for(0..self.blocks.depth) |j| {
+				for(0..self.blocks.height) |z| {
+					const block = self.blocks.get(i, j, z);
+
+					const x: f32 = @floatFromInt(i);
+					const y: f32 = @floatFromInt(j);
+
+					var newX: i64 = @intFromFloat(@round((x + 1)*cos - (y + 1)*sin));
+					var newY: i64 = @intFromFloat(@round((x + 1)*sin + (y + 1)*cos));
```

Comment:
Using floats for this seems needlessly complicated.
Please just manually do the coordinate swapping, it's not that hard:
```
if(angle == 90 or angle == 270) swap coordinates and invert one
if(angle == 180) invert both coordinates
inverting => pos = len - 1 - pos
```
Also this code should be done outside the z loop.

Reply:
> Also this code should be done outside the z loop.

I would bet that optimizer is able to notice that those variables do not depend on z variable and will move this outside of the z loop automagically.

Reply:
This isn't really about the optimizer. I just prefer having the z loop nice and simple.